MoreLinq.Test/FallbackIfEmptyTest.cs

```diff
@@ -39,6 +39,28 @@ public void FallbackIfEmptyWithEmptySequence()
         }
 
         [Test]
+        public void FallbackIfEmptyPreservesSourceCollectionIfPossible()
+        {
+            var source = new int[] { 1 };
+            // ReSharper disable PossibleMultipleEnumeration
+            Assert.AreSame(source.FallbackIfEmpty(12), source);
```

Comment:
I agree that ccca164c20d889716ad54e22b75275e6beb24c4b isn't any better. What I had more in mind was this:

```c#
[TestCase(new[] { 12                     })]
[TestCase(new[] { 12, 23                 })]
[TestCase(new[] { 12, 23, 34             })]
[TestCase(new[] { 12, 23, 34, 45         })]
[TestCase(new[] { 12, 23, 34, 45, 56     })]
[TestCase(new[] { 12, 23, 34, 45, 56, 67 })]
public void FallbackIfEmptyPreservesSourceCollectionIfPossible(int[] fallback)
{
    var source = new[] { 1 };
    Assert.AreSame(source.FallbackIfEmpty(fallback), source);
}
```